src/vaults/PoolManager.sol

```diff
@@ -96,13 +98,20 @@ contract PoolManager is Auth, Recoverable, IPoolManager, IUpdateContract, IPoolM
 
     // --- Outgoing message handling ---
     /// @inheritdoc IPoolManager
+    // TODO: change to use address token as input
     function transferShares(uint16 centrifugeId, uint64 poolId, bytes16 scId, bytes32 receiver, uint128 amount)
         external
-        auth
+        payable
     {
         IShareToken shareToken_ = IShareToken(shareToken(poolId, scId));
-        shareToken_.burn(msg.sender, amount);
+        require(
```

Comment:
This allows hooks to differentiate cross-chain transfers. It means for `RestrictedTransfers`, each chain ID needs to be added as an allowed destination.